docs: add governance doc and members list (#3)
e18e/.github: GOVERNANCE.md

**Who can vote:** Core members, Advisors, and above. All Maintainers are encouraged to discuss and voice their opinion in the pull request discussion. Core members and Advisors should take the opinions of Maintainers into consideration when voting.

1. The pull request discussion thread is used to discuss the governance change.
2. The normal 3 day voting & discussion window begins with either the PR creation or the removal of `WIP:` from the PR title if the PR was created as a draft.
3. Voting can be done in the pull request via a review of either **Approve (For)** or **Change Requested (Against)**.
4. The vote must receive a simple majority (50%+) to pass.
5. **If the vote passes:** the PR is merged and the changes take effect immediately.
6. **If the vote fails:** the PR is closed and no change occurs.

## Moderation

Outlined below is the process for Code of Conduct violation reviews.

### Reporting

Anyone may report a violation. Violations can be reported in the following ways:

- In private, via email to one or more stewards.

> - In private, via direct message to a steward on Discord (e.g. [@43081j](https://github.com/43081j)).

Reviewer: Should your link here be on the first bullet point since it's a github link? Then maybe you can put your discord username here instead.

Alternatively, we could link to MEMBERS.md directly.

Author: i've updated to link to MEMBERS 👍

The **Advisor** role is available to community members who have just as large an impact as core members, but are not necessarily leading any initiatives or projects.

An advisor is often someone who is not as involved day-to-day as a core member, but is fundamental to the community's health and direction.

With the core team, advisors are deeply involved in direction of the community, external relationships, and other high-level initiatives. They are often involved in discussions around the future of the community and its projects.

#### Nomination

- To be nominated, a nominee is expected to already be performing some of the responsibilities of an Advisor.
- You can be nominated by any existing Core member, Advisor, or above
- Once nominated, there will be a vote by existing Core members and Advisors.
- See [vote rules & requirements](#voting) for info on how the vote works.

### Steward

The **Steward** is an additional role bestowed to 1 (or more) members of the community.

The role of Steward is mainly an administrative one. Stewards control and maintain sensitive community assets, assist in resolving conflicts, and act as tiebreakers in the event of disagreements.

> In some cases, a Steward can act unilaterally when they believe it is in the community's best interest and can prove that the issue cannot be resolved through normal governance procedure. The steward must publicly state their reason for unilateral action before taking it.

Reviewer 2: should some examples be provided here?

Author: i'm not sure there are any. this exists for the unknowns really, i.e. what happens when we're stuck and something fundamental needs to change.

#### Nomination

- Stewards cannot be self-nominated.

> - Only Core members and Advisors are eligible.

Reviewer 2: > Only Core members and Advisors are eligible.

Why advisors but not maintainers? I guess advisor is "higher" ranked than maintainers?

Author: advisors and core are the same "level". all of these roles are nominated by the existing people in that role.

for example, without this it means maintainers could vote someone into core when all of core members are away (i.e. none vote).

# Governance Playbook

## Voting

Certain community decisions (like governance changes and membership nominations) require a vote. Below are the changes that require a vote, and the rules that govern that vote.

A Steward may initiate a vote for any unlisted decision. [General Rules](#general-voting-rules) will apply, along with any addition rules provided at the steward's discretion. If this unlisted decision is expected to be repeated in the future, voting rules should be agreed on and then added to this document.

### General Voting Rules

- Members may abstain from any vote.
- Members who do not vote within 3 days will automatically abstain.
- Stewards may reduce the 3 day automatic abstain for urgent decisions.
- Stewards reserve the right to veto approval with a publicly disclosed reason.
- A vote requires at least 3 non-abstaining votes to be valid (or all eligible voters if there are fewer than 3). If there are not enough votes, the vote will be invalid and the decision will be deferred until a valid vote can be held.

## Voting: New Roles

This process kicks off once a valid nomination has been made.

> The group the person is being nominated for will be responsible for voting on the nomination. Anyone in a role above that group in responsibility can also vote.

Reviewer 2: > Anyone in a role above that group in responsibility can also vote.

I'm not sure the doc covers the hierarchy of roles?

Some maintainers maintain projects, while others may just contribute widely across the community.

#### Nomination

- To be nominated, a nominee is expected to already be supporting the community in some of the ways described above.
- You can be nominated by any existing Maintainer or above
- Once nominated, there will be a vote by existing maintainers
- See [vote rules & requirements](#voting) for info on how the vote works.

### Core

The **Core** role is available to community members who have a larger-than-usual impact on the community. They are seen as leaders and are listened to by the wider community, often before they have even taken on this role.

Core members are those who have significant positive impact on the community and are involved in the long-term health of it. They are usually involved in leadership of one or more community initiatives and, in most cases, build a specialism for one area over time.

#### Nomination

- To be nominated, a nominee is expected to already be performing some of the responsibilities of a Core member.
- You can be nominated by any existing Core member, Advisor, or above

> - Once nominated, there will be a vote by existing Core members.

Reviewer 2: Is it intentional that advisors can't vote on core appointments?

Author: they can

Reviewer 2: they can nominate but not vote on actual appointment

Author: they can vote. you probably didn't see i changed the wording. but it was always intended to be that way, i just missed this one out at the time

> - Once nominated, there will be a vote by existing Core members.

Author: ```suggestion
- Once nominated, there will be a vote by existing Core/Advisor members.
```

> - Once nominated, there will be a vote by existing Core/Advisor members.

Author: ```suggestion
- Once nominated, there will be a vote by existing Core members and Advisors.
```

#### Nomination

- To be nominated, a nominee is expected to already be performing some of the responsibilities of an Advisor.
- You can be nominated by any existing Core member, Advisor, or above
- Once nominated, there will be a vote by existing Core members and Advisors.
- See [vote rules & requirements](#voting) for info on how the vote works.

### Steward

The **Steward** is an additional role bestowed to 1 (or more) members of the community.

The role of Steward is mainly an administrative one. Stewards control and maintain sensitive community assets, assist in resolving conflicts, and act as tiebreakers in the event of disagreements.

In some cases, a Steward can act unilaterally when they believe it is in the community's best interest and can prove that the issue cannot be resolved through normal governance procedure. The steward must publicly state their reason for unilateral action before taking it.

#### Nomination

- Stewards cannot be self-nominated.
- Only Core members and Advisors are eligible.

> - New Stewards will be added based on a unanimous vote by the existing Steward(s).

Reviewer 2: I guess there is the opportunity for a deadlock here? If we are unable to reach all the stewards - I guess we could vote to change governance if it causes a problem in the future

Author: im not sure there's any obvious solution to this. the next point says the decision will be deferred, so it'd just come back around until someone can vote.

if somehow nobody ever can, we're doomed and the remaining team can come up with new governance rules in order to somehow define a new steward.

but its such an extreme situation im not sure it makes sense to try document how it should go.

Reviewer 2: I think this is ok for now, we can just change the document if necessary - but hopefully it wouldn't ever be (not trying to jinx it lol)

> - New Stewards will be added based on a unanimous vote by the existing Steward(s).

Reviewer 2: who can nominate new stewards?

Author: stewards, that was meant to be implied by the fact only stewards can add them

> - In the event that someone is unreachable then the decision will be deferred.

Reviewer 2: which decision?

Author: the nomination

A Steward may initiate a vote for any unlisted decision. [General Rules](#general-voting-rules) will apply, along with any addition rules provided at the steward's discretion. If this unlisted decision is expected to be repeated in the future, voting rules should be agreed on and then added to this document.

### General Voting Rules

- Members may abstain from any vote.
- Members who do not vote within 3 days will automatically abstain.
- Stewards may reduce the 3 day automatic abstain for urgent decisions.
- Stewards reserve the right to veto approval with a publicly disclosed reason.
- A vote requires at least 3 non-abstaining votes to be valid (or all eligible voters if there are fewer than 3). If there are not enough votes, the vote will be invalid and the decision will be deferred until a valid vote can be held.

## Voting: New Roles

This process kicks off once a valid nomination has been made.

The group the person is being nominated for will be responsible for voting on the nomination. Anyone in a role above that group in responsibility can also vote.

General hierarchy is as follows (in order of base to top responsibility):
- Maintainer
- Core / Advisor

> - Steward

Author: ```suggestion
General hierarchy is as follows (in order of top responsibility to base responsibility):
- Steward
- Core / Advisor
- Maintainer
```

A Steward may initiate a vote for any unlisted decision. [General Rules](#general-voting-rules) will apply, along with any addition rules provided at the steward's discretion. If this unlisted decision is expected to be repeated in the future, voting rules should be agreed on and then added to this document.

### General Voting Rules

- Members may abstain from any vote.
- Members who do not vote within 3 days will automatically abstain.
- Stewards may reduce the 3 day automatic abstain for urgent decisions.
- Stewards reserve the right to veto approval with a publicly disclosed reason.
- A vote requires at least 3 non-abstaining votes to be valid (or all eligible voters if there are fewer than 3). If there are not enough votes, the vote will be invalid and the decision will be deferred until a valid vote can be held.

## Voting: New Roles

This process kicks off once a valid nomination has been made.

The group the person is being nominated for will be responsible for voting on the nomination. Anyone in a role above that group in responsibility can also vote.

General hierarchy is as follows (in order of base to top responsibility):
- Maintainer
- Core
- Advisor

> - Steward

Reviewer 2: why is it in reverse order 😄

Author: reverse order of what?

it says its in order of base to top responsibility, which it is, in forward order (not reverse).

can you explain a bit more?

Reviewer 2: my brain skipped over the stuff in brackets and just saw the list and was like, wait why is maintainer the highest role XD

Reviewer: It also feels reversed to me even though the wording explains it. I think it's more common for "hierarchy" to be conveyed top to bottom.

> - Steward

Author: ```suggestion
- Maintainer
- Core / Advisor
- Steward
```
